src/ds3.c

```diff
@@ -121,7 +122,7 @@ void ds3_client_register_logging(ds3_client* client, ds3_log_lvl log_lvl, void (
 }
 
 ds3_str* ds3_str_init(const char* string) {
-    size_t size = strlen(string);
+    size_t size = strlen(string)+1;
```

Comment:
Was there a probable when were were not adding 1 for the `null` character?  `g_strndup` which is used in `ds3_str_init_with_size` already accounts for the space for the `null` character.